### PR TITLE
Commander: always allow to switch to LAND mode, also if requirements are not met

### PR DESCRIPTION
**Pull request by sfuhrer**
### Solved Problem
The manual trigger of "Land" mode is often done in emergencies, when you want the vehicle to stop the current operation and land ("come down") at the current spot. As it is though an auto flight mode, the Land mode gets rejected when e.g. local position is invalid. In these scenarios the only option for the pilot is to land the vehicle manually, which is a more complex action than clicking on a single button. And for VTOL there comes additional complexity should the pilot have to land it manually, as in that case he/she has to also trigger the transition manually. If in Land/Descend mode the transition to Hover is automatically done.

### Solution
For now hard-coded that the switching into Land does never get rejected. We maybe want to hide it behind a param or an extra field in the mavlink messages (see #Alternatives). 

### Changelog Entry
For release notes:
```
Feature: Commander: always allow to switch to LAND mode, also if requirements are not met
```

### Alternatives
- we could make a parameter "always allow to manually switch to Land mode" and link the forcing of the mode to it
- add "force" flag to mavlink messages MAV_CMD_DO_SET_MODE and MAV_CMD_NAV_LAND

### Context
Screenrecording of simulation of targeted use case:
- VTOL flying GPS denied, dead-reckons for a while until position accuracy too low (COM_POS_FS_EPH)
- to prevent failsafe action, user switches to manual Altitude mode
- to land vehicle, user click on "Land" button --> mode is not rejected but system immediately switches to the Land failsafe, which is Descend

Note: requires this fix: https://github.com/PX4/PX4-Autopilot/pull/23578

https://github.com/user-attachments/assets/84fe6e26-ba7a-42ab-8876-56240f470496

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -875,7 +875,16 @@ Commander::handle_command(const vehicle_command_s &cmd)
 			}
 
 			if (desired_nav_state != vehicle_status_s::NAVIGATION_STATE_MAX) {
-				if (_user_mode_intention.change(desired_nav_state, getSourceFromCommand(cmd))) {
+				bool force = false;
```
Comment (RomanBapst) [inline src/modules/commander/Commander.cpp:878]:
```suggestion
				const bool force = desired_nav_state == vehicle_status_s::NAVIGATION_STATE_AUTO_LAND;
```

**Comment by KonradRudin** [comment]
This is only needed because QGC cannot distinguish between LAND and Descend and we can't set DESCEND right? because the proper way would be to actually try to switch to the descend mode directly from AMC, right?

**Comment by sfuhrer** [comment]
> This is only needed because QGC cannot distinguish between LAND and Descend and we can't set DESCEND right? because the proper way would be to actually try to switch to the descend mode directly from AMC, right?

Basically yes, though it's not only in QGC where the interface for switching into Descend is missing, but also in Mavlink and PX4 (where the state machine is built up such that it can only be entered as a fall-back flight flight).

My argument with this PR is that the mayor intent of a pilot that is triggering a "Land" is to "come down roughly at the current position", which is still fulfilled by Descend.

File: src/modules/commander/Commander.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/commander/Commander.cpp]: thanks, much neater

Comment (sfuhrer) [reply src/modules/commander/Commander.cpp]: applied it

**Review by KonradRudin — APPROVED** [review]

**Pull request merged**